tests/bgp/test_bgp_policy.py

```diff
@@ -0,0 +1,554 @@
+              "TWO_PREFIX_RM_V4", "TWO_PREFIX_V4", "2")
+    verify_rm(gather_info['duthost'], gather_info['neigh_ip_v6'], gather_info['cli_options'], "ipv6",
+              "TWO_PREFIX_RM_V6", "TWO_PREFIX_V6", "2")
+
+    # remove two prefix route map
+    cmd = 'vtysh{} -c "config" -c "router bgp {}" -c "address-family ipv4 unicast" \
+           -c "no neighbor {} route-map TWO_PREFIX_RM_V4 out" \
+           -c "address-family ipv6 unicast" -c "no neighbor {} route-map TWO_PREFIX_RM_V6 out"' \
+          .format(gather_info['cli_options'], gather_info['dut_asn'], gather_info['neigh_ip_v4'],
+                  gather_info['neigh_ip_v6'])
+    logger.debug(gather_info['duthost'].shell(cmd, module_ignore_errors=True))
+    logger.debug(gather_info['duthost'].shell('vtysh{} -c "clear bgp * soft"'.format(gather_info['cli_options']),
+                                              module_ignore_errors=True)['stdout'])
+    time.sleep(10)
+    check_baseline(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], num_v4_routes,
+                   num_v6_routes)
+
+    # Configure and verify policy with AS-path prepend
+    cmd = 'vtysh{} -c "config" -c "router bgp {}" -c "route-map TWO_PREFIX_RM_V4 permit 10" -c \
+        "set as-path prepend 12345"'.format(gather_info['cli_options'], gather_info['dut_asn'])
+    logger.debug(gather_info['duthost'].shell(cmd, module_ignore_errors=True))
+    cmd = 'vtysh{} -c "config" -c "router bgp {}" -c "route-map TWO_PREFIX_RM_V6 permit 10" -c \
+        "set as-path prepend 12345"'.format(gather_info['cli_options'], gather_info['dut_asn'])
+    logger.debug(gather_info['duthost'].shell(cmd, module_ignore_errors=True))
+
+    cmd = 'vtysh{} -c "config" -c "router bgp {}" -c "address-family ipv4 unicast" \
+           -c "neighbor {} route-map TWO_PREFIX_RM_V4 out" \
+           -c "address-family ipv6 unicast" -c "neighbor {} route-map TWO_PREFIX_RM_V6 out"' \
+          .format(gather_info['cli_options'], gather_info['dut_asn'], gather_info['neigh_ip_v4'],
+                  gather_info['neigh_ip_v6'])
+    logger.debug(gather_info['duthost'].shell(cmd, module_ignore_errors=True))
+    logger.debug(gather_info['duthost'].shell('vtysh{} -c "clear bgp * soft"'.format(gather_info['cli_options']),
+                                              module_ignore_errors=True)['stdout'])
+    time.sleep(10)
+
+    verify_rm(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['cli_options'], "ipv4",
+              "TWO_PREFIX_RM_V4", "TWO_PREFIX_V4", "2")
+    verify_rm(gather_info['duthost'], gather_info['neigh_ip_v6'], gather_info['cli_options'], "ipv6",
+              "TWO_PREFIX_RM_V6", "TWO_PREFIX_V6", "2")
+
+    # verify '12345' in AS Path
+    output = gather_info['duthost'].shell('show ip bgp neighbors {} advertised-routes'.format(
+        gather_info['neigh_ip_v4']))['stdout']
```

Comment:
this is too wild. we can get the specific routes by using 'show ip bgp neighbors <neigh_ip> advertised-routes | grep <route-prefix>' so there is no need to check ASN in the whole output.

Reply:
I don't understand what you are asking.  Are you suggesting using grep to find the ASN instead of asserting the ASN is in the output as a whole?

Reply:
Sorry, the github folded the code block in my comment. I suggest using grep to extract only relevant routes by 'show ip bgp neighbors NIEGH-IP advertised-routes | grep ROUTE-PREFIX' and then assert ASN in those routes.

Reply:
I am still confused as to what you are trying to fix here.  The current code and your suggestion both run the same command.  The grep would mean the output would be a number of lines equal to the number of routes, but both styles are then looking for the ASN in the output.  Is there a use case where the ASN would be in the full advertised-routes output, but be on an incorrect route?

Reply:
I haven't seen such a case by myself. However, this test may be used by people on various testbeds across different companies after it got merged. We cannot make an assumption on how people configure their testbed. The test should be robust enough to work in the case where the 12345 ASN is attached to routes not relevant in this test.

Reply:
I understand and have adjusted the command accordingly.